### ISSUE TITLE
Can't compile wLE starting with commit: b950488

### ISSUE THREAD

**Issue by Jay-Jay-OPL** [AKuHAK/uLaunchELF#20]
The last version of wLE that I can compile is commit: https://github.com/AKuHAK/uLaunchELF/commit/763495f1ccb15a756cf6df801f73f90331107f8f (Aug 27, 2017)

But then that got broken starting with this commit: https://github.com/AKuHAK/uLaunchELF/commit/b95048872ab451ba71f5d8ca26bc655d53de8447 (Feb 18, 2018)

... and pretty much to any commit after that, gives me the same general error -- even the most current commit: https://github.com/AKuHAK/uLaunchELF/commit/b4f8a48b561e5f11cecb31790f3c834c27f86583 (Aug 11, 2018 )

I get the following error at the end of the compiling:

```
In file included from launchelf.h:8,
                 from main.c:4:
githash.h:1:20: warning: extra tokens at end of #ifndef directive
githash.h:2:17: warning: extra tokens at end of #ifndef directive
githash.h:2:1: unterminated #ifndef
githash.h:1:1: unterminated #ifndef
main.c: In function `Show_About_uLE':
main.c:296: `ULE_VERDATE' undeclared (first use in this function)
main.c:296: (Each undeclared identifier is reported only once
main.c:296: for each function it appears in.)
main.c:298: `GIT_HASH' undeclared (first use in this function)
make: *** [main.o] Error 1
```

Unless, I am using the wrong command?  I use the `make rebuild` command. -- Please advise?

**UPDATE**:

If I revert only the changes made to this commit: https://github.com/AKuHAK/uLaunchELF/commit/b95048872ab451ba71f5d8ca26bc655d53de8447 (Feb 18, 2018)

I can then compile the latest wLE version: https://github.com/AKuHAK/uLaunchELF/commit/b4f8a48b561e5f11cecb31790f3c834c27f86583 (Aug 11, 2018 )

But it won't have the data that is generated inside the About page of wLE. :-(

**Comment by sp193** [comment]
Do you have git installed? You need it to compile LaunchELF.

**Comment by Jay-Jay-OPL** [comment]
Yes, I believe so, unless it's a special GIT?  Keep in mind I use the Windows Compiling Environment.

Though I use the **msys** window (`C:\MinGW\msys\1.0\msys.bat`)--since I prefer it, but I do have GIT, Mercurial, and MinGW.

Are you referring to any of the following?

1. Git Bash
2. Git CMD
3. Git GUI

I don't use those, since I prefer **msys**, but let me know if that is what I am suppose to use?

Please advise?

**Comment by Jay-Jay-OPL** [comment]
For both commands, I get the following message:

```
JJ@PS2HOME ~/wle
$ git show -s --format=%cd --date=local
Sat Aug 11 08:03:14 2018

JJ@PS2HOME ~/wle
$ git show -s --format=%cd --date=local | tr -d "\n"
Sat Aug 11 08:03:14 2018
```

If you like to know what I am using as a windows compiling environment, here is the guide:   http://www.ps2-home.com/forum/viewtopic.php?f=50&t=169

So I am not use to using GIT Bash window or the other ones I listed above under GIT.  

I use msys that is under mingw.  Because when I installed ps2toolchain, it is under the `C:/MinGW` directory and not under the `C:/GIT` directory -- like that guide for Windows Compiling Guide goes.  -- FYI: that guide was given to me by @doctorxyz 

I am currently wondering if I should install everything again, but through the GIT BASH cmd window so it's under the GIT directory and not under the mingw directory??  Since I notice that GIT for Windows now has a directory like this: `C:\Git\mingw64`

**Comment by sp193** [comment]
How strange. It seems like it should work, but yet it does not work in practice....

Please do not use the Git Bash environment. It is another MSYS environment, which may not have the necessary libraries and paths set up.

When building LaunchELF fails, is there a githash.h file? Please paste its contents here.

**Comment by sp193** [comment]
It looks like the newline escape characters (\n) is not being expanded on.

From https://superuser.com/questions/256465/echo-newline-character-not-working-in-bash, it seems like the expansion of escape characters is implementation-dependant.

Can you try the following, so we can tell whether there is a way to make this work:
- echo "Hello\nWorld"
- echo -e "Hello\nWorld"
- echo -e -n "Hello\World"
- printf "Hello\nWorld"

The correct output is:
``` Hello
World```

**Comment by Jay-Jay-OPL** [comment]
Okay, here you go:

```
JJ@PS2HOME ~
$ echo "Hello\nWorld"
Hello\nWorld

JJ@PS2HOME ~
$ echo -e "Hello\nWorld"
Hello
World

JJ@PS2HOME ~
$ echo -e -n "Hello\nWorld"
Hello
World

JJ@PS2HOME ~
$ printf "Hello\nWorld"
Hello
World
```

**Comment by sp193** [comment]
Thanks. Fixed in commit 3576a2b.
